File: JS/Math/ReverseInteger.js
module.exports = {
	//param A : integer
	//return an integer
	reverse : function(A){
        var sign = 1;
        var MAX = 2147483647;
        if(A < 0){
            A = -A;
            sign = -1;
        }
        var res = 0;
        while(A > 0){
            //console.log(A+" "+res);
            res = res * 10 + A % 10;
            if(res > MAX)return 0;
            A = parseInt(A / 10);
        }
        return sign * res;
	}
};

File: Java/Math/ReverseInteger.java
public class Solution {
	public int reverse(int a) {
	    int sign = 1;
	    if(a < 0){
	        sign = -1;
	        a = -a;
	    }
	    long reverse = 0;
	    while(a > 0){
	        reverse = reverse * 10 + a % 10;
	        //System.out.println(a+" "+reverse);
	        a /= 10;
	        if(reverse > Integer.MAX_VALUE)return 0;
	    }
	    return (int)(sign*reverse);
	}
}
